src/main/java/com/example/controllers/ActivityController.java

```diff
@@ -0,0 +1,120 @@
+	@Autowired
+	CreditAccountDao creditaccountDao;
+
+	@Autowired
+	ActivityDao activityDao;
+
+	Logger logger = LoggerFactory.getLogger(ActivityController.class);
+
+	@RequestMapping
+	public String activity(final Model model, final Principal principal) {
+
+		CashAccount cashAccount = new CashAccount();
+		Account account = accountDao.findUsersByUsername(principal.getName()).get(0);
+		List<CashAccount> cashAccounts = cashaccountDao.findCashAccountsByUsername(principal.getName());
+
+		List<Transaction> firstCashAccountTransfers = activityDao.findTransactionsByCashAccountNumber(cashAccounts.get(0).getNumber());
+		List<Transaction> reverseFirstCashAccountTransfers = Lists.reverse(firstCashAccountTransfers);
+
+		model.addAttribute("account", account);
+		model.addAttribute("cashAccounts", cashAccounts);
+		model.addAttribute("cashAccount", cashAccount);
+		model.addAttribute("firstCashAccountTransfers", reverseFirstCashAccountTransfers);
+		model.addAttribute("actualCashAccountNumber", cashAccounts.get(0).getNumber());
+
+		return "accountActivity";
+	}
+
+	@RequestMapping("{account}/detail")
+	public String viewActivityByNumber(@PathVariable("account") final String number, final Model model, final Principal principal) {
+
+		CashAccount cashAccount = new CashAccount();
+		Account account = accountDao.findUsersByUsername(principal.getName()).get(0);
+		List<CashAccount> cashAccounts = cashaccountDao.findCashAccountsByUsername(principal.getName());
+
+		List<Transaction> firstCashAccountTransfers = activityDao.findTransactionsByCashAccountNumber(number);
+		List<Transaction> reverseFirstCashAccountTransfers = Lists.reverse(firstCashAccountTransfers);
+
+		model.addAttribute("account", account);
+		model.addAttribute("cashAccounts", cashAccounts);
+		model.addAttribute("cashAccount", cashAccount);
+		model.addAttribute("firstCashAccountTransfers", reverseFirstCashAccountTransfers);
+		model.addAttribute("actualCashAccountNumber", number);
+
+		return "accountActivity";
+	}
+
+	@RequestMapping("credit")
+	public String viewActivityByCreditNumber(@RequestParam("number") final String number, final Model model, final Principal principal) {
+
+		logger.info("Showing info for credit card:" + number);
+
+		Account account = accountDao.findUsersByUsername(principal.getName()).get(0);
+
+		model.addAttribute("account", account);
+		model.addAttribute("actualCreditCardNumber", number);
+
+		return "creditActivity";
+	}
+
+	@RequestMapping(value = { "", "/activity", "/detail" }, method = RequestMethod.POST)
+	public String changeAccount(@Valid @ModelAttribute final CashAccount cashAccount, final BindingResult bindingResult, final Model model,
+			final Principal principal) {
+
+		String postNumber = cashAccount.getNumber();
+
+		CashAccount cashAccountNewForm = new CashAccount();
+		Account account = accountDao.findUsersByUsername(principal.getName()).get(0);
+		List<CashAccount> cashAccounts = cashaccountDao.findCashAccountsByUsername(principal.getName());
```

Comment:
#### Automated PR Comment From Polaris SAST
<!-- 0c42fb8bef82967132ab7be54dff199d -->
<!-- Comment managed by Git Commenter, do not modify! -->
### Polaris SAST Issue - SQL Injection
_High CWE-89_
Untrusted user-supplied data is inserted into a SQL statement without adequate validation, escaping, or filtering.

A user can change the intent of the SQL query, which may inappropriately disclose or corrupt data within the database.

### How to fix
Rewrite all SQL queries constructed through dynamic concatenation to use an injection-safe query mechanism such as prepared statements with parameterized queries.

Most modern programming languages provide a feature called "parameterized queries" that allow user-supplied data to be inserted safely as values in dynamic SQL queries. Rather than construct the dynamic SQL query by concatenating user-supplied data to static SQL query string fragments, data values are identified in the query by parameter markers or variables. Dynamic data is then passed through a mechanism provided by SQL that prevents the supplied data from changing the meaning of the query.

Note: the exact syntax and use of prepared statements with parameterized queries vary from language to language.